bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetThermoregulationHandler.java

```diff
@@ -0,0 +1,375 @@
+                            currentFunction));
+                } catch (MalformedFrameException | OWNException e) {
+                    logger.warn("handleSetpointCommand() {}", e.getMessage());
+                }
+            }
+        } else {
+            logger.warn("handleSetpointCommand() Unsupported command {} for thing {}", command, getThing().getUID());
+        }
+    }
+
+    private void handleMode(Command command) {
+        if (command instanceof StringType) {
+            Where w = deviceWhere;
+            if (w != null) {
+                try {
+                    OPERATION_MODE mode = OPERATION_MODE.valueOf(((StringType) command).toString());
+                    if (mode == OPERATION_MODE.MANUAL) {
+                        logger.debug("handleMode() mode={} function={} setPointTemp={}°", mode.toString(),
+                                currentFunction.toString(), currentSetPointTemp);
+                    } else {
+                        logger.debug("handleMode() mode={} function={}", mode.toString(), currentFunction.toString());
+                    }
+                    send(Thermoregulation.requestWriteMode(w.value(), mode, currentFunction, currentSetPointTemp));
+                } catch (OWNException e) {
+                    logger.warn("handleMode() {}", e.getMessage());
+                } catch (IllegalArgumentException e) {
+                    logger.warn("handleMode() Unsupported command {} for thing {}", command, getThing().getUID());
+                    return;
+                }
+            }
+
+        } else {
+            logger.warn("handleMode() Unsupported command {} for thing {}", command, getThing().getUID());
+        }
+    }
+
+    private void handleFunction(Command command) {
+        logger.debug("handleFunction() (command={})", command);
+        if (command instanceof StringType) {
+            Where w = deviceWhere;
+            if (w != null) {
+                try {
+                    FUNCTION function = FUNCTION.valueOf(((StringType) command).toString());
+                    logger.debug("handleFunction() mode={}", function.toString());
+                    send(Thermoregulation.requestWriteFunction(w.value(), function));
+                } catch (OWNException e) {
+                    logger.warn("handleFunction() {}", e.getMessage());
+                } catch (IllegalArgumentException e) {
+                    logger.warn("handleFunction() Unsupported command {} for thing {}", command, getThing().getUID());
+                    return;
+                }
+            }
+        } else {
+            logger.warn("handleFunction() Unsupported command {} for thing {}", command, getThing().getUID());
+        }
+    }
+
+    private QuantityType<Temperature> commandToQuantityType(Command command, Unit<Temperature> unit) {
+        return new QuantityType<Temperature>(command.toFullString());
+    }
```

Comment:
What's the purpose of this method?

Reply:
It was propedeutic for other stuffs but currently it is without sense. I've removed it.